lib/money/parser/simple.rb

```diff
@@ -15,8 +17,8 @@ def parse(input, currency, strict: false)
           currency = Money::Helpers.value_to_currency(currency)
           return unless currency
 
-          amount = BigDecimal(input, exception: false)
-          if amount
+          coerced = input.to_s
+          if SIGNED_DECIMAL_MATCHER.match?(coerced) && (amount = BigDecimal(coerced, exception: false))
             Money.new(amount, currency)
```

Comment:
nit: we should avoid using an assignment on the second part of the boolean

Reply:
Our style guide [allows it](https://ruby-style-guide.shopify.dev/#:~:text=wrap%20assignment%20in%20parentheses%20when%20using%20its%20return%20value%20in%20a%20conditional%20statement.). I'll look into adding our Rubocop rules to this repo because linters can do this better than humans 😄

Reply:
This will be a fair bit of work: https://github.com/Shopify/money/pull/240 will circle back after this and some other outstanding PRs are merged to minimize disruption.